near-contract-standards/src/storage_management/mod.rs

```diff
@@ -0,0 +1,56 @@
+use near_sdk::json_types::{ValidAccountId, U128};
+use near_sdk::serde::Serialize;
+
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct StorageBalance {
+    pub total: U128,
+    pub available: U128,
+}
+
+#[derive(Serialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct StorageBalanceBounds {
+    pub min: U128,
+    pub max: Option<U128>,
+}
+
+pub trait StorageManagement {
+    // if `registration_only=true` MUST refund above the minimum balance if the account didn't exist and
+    //     refund full deposit if the account exists.
+    fn storage_deposit(
+        &mut self,
+        account_id: Option<ValidAccountId>,
+        registration_only: Option<bool>,
+    ) -> StorageBalance;
+
+    /// Withdraw specified amount of available Ⓝ for predecessor account.
+    ///
+    /// This method is safe to call. It MUST NOT remove data.
+    ///
+    /// `amount` is sent as a string representing an unsigned 128-bit integer. If
+    /// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+    /// predecessor account's available balance, contract MUST panic.
+    ///
+    /// If predecessor account not registered, contract MUST panic.
```

Comment:
```suggestion
    /// If predecessor account not registered, contract MUST panic.
    ///
    /// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted
    /// function-call access-key call (UX wallet security)
```

As added in https://github.com/near/NEPs/pull/166/commits/beb30876f6027ada4db0cf5a63de9aa4e3e086b9